src/core/load_balancing/rls/rls.cc

```diff
@@ -512,6 +512,15 @@ class RlsLb final : public LoadBalancingPolicy {
       // Moves entry to the end of the LRU list.
       void MarkUsed() ABSL_EXCLUSIVE_LOCKS_REQUIRED(&RlsLb::mu_);
 
+      // Clears child_policy_wrappers_. The returned list should be destroyed
+      // after releasing the lock.
+      std::vector<RefCountedPtr<ChildPolicyWrapper>> ClearChildPolicyWrappers()
```

Comment:
Suggest calling this `TakeChildPolicyWrappers()`.

Also, it looks like all of the callers are going to need to append the entries in the returned vector to another vector in which they are accumulating results.  Given that, I suggest just passing in the vector being used to accumulate the results and having this method append the entries directly.  That will be both more ergonomic for the callers and will eliminate an unnecessary allocation.

Reply:
Since we are returning the existing `child_policy_wrappers_` with std::move, we shouldn't be allocating any new memory for this. I'll take the suggestion since it does seem more ergonomic even though it goes against the style guide :D .

Reply:
Oh, good point about it not actually saving an allocation.  But yeah, I do think it's more ergonomic this way anyway.